### ISSUE TITLE
Ubuntu Compilation Issues

### ISSUE THREAD

**Issue by jtramm** [openmc-dev/openmc#1070]
I'm running into two compilation issues on Ubuntu 16.04 LTS with gcc 5.4:

- The first issue occurs on line 125 nuclide_header.F90:

```
[ 16%] Building Fortran object CMakeFiles/libopenmc.dir/src/nuclide_header.F90.o
/home/jtramm/openmc/src/nuclide_header.F90:125:64:

   real(8), parameter :: CACHE_INVALID = dble(Z"FFE0000000000000")
                                                                1
Error: Integer too big for integer kind 8 at (1)
/home/jtramm/openmc/src/nuclide_header.F90:852:46:

     micro_xs % elastic         = CACHE_INVALID
                                              1
Error: Symbol ‘cache_invalid’ at (1) has no IMPLICIT type
CMakeFiles/libopenmc.dir/build.make:782: recipe for target 'CMakeFiles/libopenmc.dir/src/nuclide_header.F90.o' failed
make[3]: *** [CMakeFiles/libopenmc.dir/src/nuclide_header.F90.o] Error 1
CMakeFiles/libopenmc.dir/build.make:798: recipe for target 'CMakeFiles/libopenmc.dir/src/nuclide_header.F90.o.provides' failed
make[2]: *** [CMakeFiles/libopenmc.dir/src/nuclide_header.F90.o.provides] Error 2
CMakeFiles/Makefile2:142: recipe for target 'CMakeFiles/libopenmc.dir/all' failed
make[1]: *** [CMakeFiles/libopenmc.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

When I change the ```dble(Z"FFE0000000000000")``` to ```-1``` it seems to compile fine (though not sure if this is a good idea).

- I then run into a second issue later on during linking:

```
[ 99%] Built target libopenmc
Scanning dependencies of target openmc
[ 99%] Building CXX object CMakeFiles/openmc.dir/src/main.cpp.o
[100%] Linking CXX executable bin/openmc
lib/libopenmc.so: undefined reference to `openmc::H5TypeMap<unsigned int>::type_id'
collect2: error: ld returned 1 exit status
CMakeFiles/openmc.dir/build.make:99: recipe for target 'bin/openmc' failed
make[2]: *** [bin/openmc] Error 1
CMakeFiles/Makefile2:182: recipe for target 'CMakeFiles/openmc.dir/all' failed
make[1]: *** [CMakeFiles/openmc.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by paulromano** [comment]
In theory it *shouldn't* be a problem! I think it's worth getting fixed -- do you want to take a shot at a PR to fix these two issues? For the `CACHE_INVALID` one, setting to an arbitrary negative value should probably be ok. The other option would be to move the definition of `CACHE_INVALID` to nuclide.h/nuclide.cpp and then add a binding on the Fortran side.

**Comment by paulromano** [comment]
Closed with the merging of #1071
